core/src/main/scripts/importer/cbioportal_common.py

```diff
@@ -638,90 +639,101 @@ def parse_metadata_file(filename,
                     {True: 'case list', False: 'meta'}[case_list],
                     extra={'filename_': filename,
                            'line_number': line_index + 1})
-                meta_file_type = None
-                return metaDictionary, meta_file_type
+                meta_dictionary['meta_file_type'] = None
+                return meta_dictionary
             key_value = line.split(':', 1)
             if len(key_value) == 2:
-                metaDictionary[key_value[0]] = key_value[1].strip()
+                meta_dictionary[key_value[0]] = key_value[1].strip()
 
+    # Determine meta file type
     if case_list:
         meta_file_type = MetaFileTypes.CASE_LIST
     else:
-        meta_file_type = get_meta_file_type(metaDictionary, logger, filename)
-        # if type could not be inferred, no further validations are possible
-        if meta_file_type is None:
-            return metaDictionary, meta_file_type
+        meta_file_type = get_meta_file_type(meta_dictionary, logger, filename)
+
+    meta_dictionary['meta_file_type'] = meta_file_type
```

Comment:
I disagree with the identation change. I think it should remain, and this line `meta_dictionary['meta_file_type'] = meta_file_type` should indeed follow after the if/else. The reason is that `if meta_file_type is None:` only applies to the output of `get_meta_file_type`, so keep it close to that....otherwise you risk making the code more confusing if you put related things far apart.